pkg/substitution/substitution.go

```diff
@@ -76,7 +76,8 @@ func ValidateVariableIsolated(name, value, prefix, contextPrefix, locationName,
 // Extract a the first full string expressions found (e.g "${input.params.foo}"). Return
 // "" and false if nothing is found.
 func extractExpressionFromString(s, prefix string) (string, bool) {
-	pattern := fmt.Sprintf("\\$({%s.(?P<var>%s)})", prefix, parameterSubstitution)
+	//TODO(1170): Regex matches both ${} and $(), we will need to remove ${} compatibility.
```

Comment:
The most minor of nits: These todos are labelled differently from others elsewhere in the diff: `TODO(1170)` vs `TODO(#1170)`.  The only reason this is noteworthy is because I assume we're going to want to grep for the pattern and remove all of them in future. Might be helpful to future us if all the todos are the same string.

Reply:
whoops, thanks for catching this! :D

Reply:
i think while im here i'm going to put that regex into a constant ... 😇